### DISCUSSION TITLE
telegram api not working on tasmota 12.3.1

### DISCUSSION THREAD

**Discussion by biotecov** [arendst/Tasmota#17689] [Q&A] [answered]
Hi,
I was using telegram (sending messages to my telegram from several devices) some time ago, and that stopped suddenly on june 2021. I did not updated devices a that time, I read) but now I was interested in getting this feature working again. I compiled tasmota 12.3.1 with #define USE_TELEGRAM and nothing else special, flashed on a Sonoff S20, defined tmchatid and tmtoken with doubl-checked correct values, and set tmstate 1; tmstate 3; tmstate 5. But tmsend just returns MQT: stat/s20e1/RESULT = {"TmSend":"Failed"}. Everything else works as expected. Can anybody confirm if this is known issue or should I do something else? I tried with an ancient Telegram Bot that worked well (and that seems to be valid otherwise), and with a newly created Telegram Bot. Both bots respond to messages sent from a browser.
Regards.

**Comment by biotecov** [comment]
As a complement, I just compiled and flashed version 10.0.0, and telegram works on that version. It is too late (3AM for me) to do more tests tonight, but it is now clear that something changed in tasmota code between v10.0.0 and v12.3.1. Maybe tomorrow night I may extend tests to bracket on which version it stopped to work.

**Comment by biotecov** [comment, marked as answer]
Hi,
after further testing, I found what is wrong ... As I said in the last post, v.10.0.0 worked out-of-the-box (after setting the USE_TELEGRAM conditional compilation directive and setting the relevant tmchatid etc...), but not v.12.3.1. After further testing, I saw that v.10.0.0 was the latest "working" version. v.10.1.0 was (apparently) "faulty". So I looked what was different between these two consecutive sub-versions.
My attention was quickly captured by this line in the changelog:

- MQTT TLS dual mode (CA or fingerprint) in same firmware, ``SetOption132 1`` to force fingerprint

So I went back to v.10.1.0 that was running on my testing device, forced the fingerprint usage typing "SetOption132 1" on the console, and tried again tmsend .... Tachan .... success !!!

Version 12.3.1 also work with this simple precaution. So telegram is functional on the latest version. The only thing wich is "faulty" is that the telegram conditional compilation directive does no activate SO132 by default, and especially, the lack of any documentation stating the necessity of doing that for telegram.

To finish, may I suggest to the people than maintains the docs to create an entry explaining how to use telegram, or at least, to mention that fact in the description of SetOption132? For example, I would suggest a change like this one:

SetOption132 | When  MQTT TLS **_or telegram_** is enabled, forces fingerprint validation of server identity  instead of checking the identify against a certificate authority (CA). **_Note that the telegram api will fail to connect if this option is not set._**
1 = Fingerprint, 0 = CA

Finally, I hope that the lack of any comment or answer to my initial request does not mean that the telegram functionality is lacking support and could be dropped ... I found it highly interesting and worth being maintained.

Regards to all.

**Reply by barbudor** [reply]
Happy you found the solution

Documentation is written by users that contributes to the project. Contributions to the doc are welcomed.
If you feel that something is missig or should be corrected, feel free to suggest an edit by clicking on the pencil button on the top right of the page.

**Reply by biotecov** [reply]
Ok, so I did regarding SO132 and telegram
